README.md

```diff
@@ -0,0 +1,143 @@
+- Allows AllocatorDaos to list operators to manage AllocatorVaults, funnels and conduits in a per-action resolution.
+- Warded by the Pause Proxy, which needs to add a new AllocatorDAO once one is onboarded.
+
+### AllocatorRegistry
+
+A registry where each AllocatorDAO’s AllocatorBuffer address is listed.
+
+### Swapper
+
+A module that pulls tokens from the AllocatorBuffer and sends them to be swapped at a callee contract. The resulting funds are sent back to the AllocatorBuffer.
+
+It enforces that:
+
+- The swap rate is not faster than a pre-configured rate.
+- The amount to swap each time is not larger than a pre-configured amount.
+- The received funds are not less than a minimal amount specified on the swap call.
+
+### Swapper Callees
+
+Contracts that perform the actual swap and send the resulting funds to the Swapper (to be forwarded to the AllocatoBuffer).
+
+- They can be implemented on top of any DEX / swap vehicle.
+- An example is SwapperCalleeUniV3, where swaps in Uniswap V3 can be triggered.
+
+### DepositorUniV3
+
+A primitive for depositing liquidity to Uniswap V3 in a fixed range. 
+
+As the Swapper, it includes rate limit protection and is designed so facilitators and automation contracts can use it.
+
+### StableSwapper
+
+An automation contract, which can be used by the AllocatorDAOs to set up recurring swaps of stable tokens (e.g NST to USDC).
+
+- In order to use it the AllocatorDAO should list it as an operator of its Swapper primitive in the AllocatorRoles contract.
+- The Swapper primitive will rate-limit the automation contract.
+
+### StableDepositorUniV3
+
+An automation contract sample, which can be used by the AllocatorDAOs to set up recurring deposits or withdraws. 
+
+- In order to use it the AllocatorDAO should list it as an operator of its DepositorUniV3 primitive in the AllocatorRoles contract.
+- The Depositor primitive will rate-limit the automation contract.
+
+### ConduitMover
+
+An automation contract sample, which can be used by the AllocatorDAOs to move funds between their AllocatorBuffer and the conduits in an automated manner.
+
+### IAllocatorConduit
+
+An interface which each Conduit should implement.
+
+## Security Model:
+- AllocatorDao Proxies can not incur a loss of more than the debt ceiling (`line`) of their respecive `ilk`.
+- A funnel operator (whether a facilitator or an automated contract) can not incur a loss of more than `cap` amount of funds per `era` interval for a specific configuration. This includes not being able to move funds directly to any unknown contract that the AllocatorDao Proxy did not approve.
+- A keeper can not incur a loss of more than the funnel opeator can, and any loss it can incur is also constrained by `req` or `req0` and `req1` for a specific configuration.
+
+## Technical Assumptions:
+- A `uint32` is suitable for storing timestamps or time intervals in the funnels, as the current version of the Allocation System is not expected to be utilized up untill 2106.
```

Comment:
```suggestion
- A `uint32` is suitable for storing timestamps or time intervals in the funnels, as the current version of the Allocation System is not expected to be utilized beyond 2106.
```

Reply:
mmm I think `beyond` 2106 implies it can be used unitl the end of 2106, which wouldn't work.

Reply:
"not expected to be used beyond 2106" is what we want to convey no? the uint32 timestamp will only be usable until `1970+2^32/3600/24/365=~2106`. I think "not expected to be used up until 2106" would mean that the system is not going to be used _before_ 2106.

Reply:
Maybe "the Allocation System is expected to be deprecated long before 2106" would be better though.

Reply:
done